code/__HELPERS/sorts/__main.dm

```diff
@@ -626,14 +626,13 @@ GLOBAL_DATUM_INIT(sortInstance, /datum/sortInstance, new())
 		var/val2 = fetchElement(L,cursor2)
 
 		while(1)
-			if(call(cmp)(val1,val2) < 0)
+			if(call(cmp)(val1,val2) <= 0)
 				if(++cursor1 >= end1)
 					break
 				val1 = fetchElement(L,cursor1)
 			else
 				moveElement(L,cursor2,cursor1)
 
-				++cursor2
```

Comment:
This extra cursor2 increment was resulting in:
[**a**,b,c,d,e] , [**a**,d,e]
cursor1=1 is at LHS a
cursor2=6 is at RHS a

RHS a is merged into the LHS just after LHS a
[**a**,a,b,c,d] , [**e**,d,e]
cursor1=1 at first LHS a
cursor2=6 at e of RHS

++cursor1
++cursor2 ++cursor2
a,[**a**,b,c,d,e,d] , [**e**]

When what we wanted was:
a,[**a**,b,c,d,e] , [**d**,e]